commit 8e54fc8910f3d131be47d434fcde15d409ae62f1
Author: Takashi Sakamoto <o-takashi@sakamocchi.jp>
Date:   Mon Sep 17 17:26:41 2018 +0900

    ALSA: fireworks: fix memory leak of response buffer at error path
    
    commit c3b55e2ec9c76e7a0de2a0b1dc851fdc9440385b upstream.
    
    After allocating memory object for response buffer, ALSA fireworks
    driver has leak of the memory object at error path.
    
    This commit releases the object at the error path.
    
    Fixes: 7d3c1d5901aa('ALSA: fireworks: delayed registration of sound card')
    Cc: <stable@vger.kernel.org> # v4.7+
    Signed-off-by: Takashi Sakamoto <o-takashi@sakamocchi.jp>
    Signed-off-by: Takashi Iwai <tiwai@suse.de>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/sound/firewire/fireworks/fireworks.c b/sound/firewire/fireworks/fireworks.c
index 71a0613d3da0..f2d073365cf6 100644
--- a/sound/firewire/fireworks/fireworks.c
+++ b/sound/firewire/fireworks/fireworks.c
@@ -301,6 +301,8 @@ do_registration(struct work_struct *work)
 	snd_efw_transaction_remove_instance(efw);
 	snd_efw_stream_destroy_duplex(efw);
 	snd_card_free(efw->card);
+	kfree(efw->resp_buf);
+	efw->resp_buf = NULL;
 	dev_info(&efw->unit->device,
 		 "Sound card registration failed: %d\n", err);
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1035_8e54fc8910f3d131be47d434fcde15d409ae62f1.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1035_8e54fc8910f3d131be47d434fcde15d409ae62f1.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
